sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/src/Response/ConnectEventResponse.cs

```diff
@@ -73,12 +81,22 @@ public ConnectEventResponse()
         /// </summary>
         /// <param name="key">State key.</param>
         /// <param name="value">State value.</param>
-        public void SetState(string key, object value)
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void SetState(string key, object value) =>
+            SetState(key, BinaryData.FromObjectAsJson(value));
+
+        /// <summary>
+        /// Set connection states.
+        /// </summary>
+        /// <param name="key">State key.</param>
+        /// <param name="value">State value.</param>
+        public void SetState(string key, BinaryData value)
         {
             // In case user cleared states.
             if (_states == null)
             {
-                _states = new Dictionary<string, object>();
+                _states = new Dictionary<string, BinaryData>();
+                States = new StringifiedDictionary(_states);
             }
             _states[key] = value;
```

Comment:
Update legacy States as well?

Reply:
We don't need to because both the legacy `States` and the new `ConnectionStates` are backed by the same `_states` dictionary.  Changes to one will automatically appear in the other.

Reply:
States is not inited by `new Stringigfied(_states)` because default `_states` is `new()` instead of `null`. Though this doesn't matter when return service cause we only repect `ConnectionStates` now. But if customer check the value after call `SetState()`. They'd find it not match for `States` and `ConnectionStates`. Shall we keep them same?

Reply:
Great catch!  The field initializer looks too much like `null`.  I fixed this for both responses.